conan/internal/api/upload.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+from conan.internal.rest.client_routes import ClientV2Router
+from conan.internal.util.files import sha1sum
+
+
+def add_urls(package_list, remote, backup_files, upload_url):
+    router = ClientV2Router(remote.url.rstrip("/"))
+    for ref, bundle in package_list.refs().items():
+        for f, fp in bundle.get("files", {}).items():
+            bundle.setdefault("urls", {})[f] = {
+                'url': router.recipe_file(ref, f), 'checksum': sha1sum(fp)
+            }
+        for pref, prev_bundle in package_list.prefs(ref, bundle).items():
+            for f, fp in prev_bundle.get("files", {}).items():
+                prev_bundle.setdefault("urls", {})[f] = {
+                    'url': router.package_file(pref, f), 'checksum': sha1sum(fp)
+                }
+
+    if upload_url:
+        url = upload_url if upload_url.endswith("/") else upload_url + "/"
+        if backup_files:
+            for file in backup_files:
+                basename = os.path.basename(file)
+                package_list.recipes.setdefault('backup_sources', {})[basename] = {
+                    'file_path': file, 'url': url + basename, 'checksum': sha1sum(file)
+                }
```

Comment:
There is one problem with the backup-sources remote.
They are really not associated to any particular remote, not to the ``remote`` argument either.
They are independent and common to all remotes. So it doesn't seem correct to put them under the ``remote`` in the json. 
But the json is structured using the remote as the first key, so this seems challenging. Maybe we should just drop the backup sources upload by URL, until it is requested.

Reply:
I agree that maybe the easiest solution for this is to drop the backup sources until someone requests it